core/src/main/scala/za/co/absa/spline/harvester/builder/write/WriteCommandExtractor.scala

```diff
@@ -54,13 +54,18 @@ class WriteCommandExtractor(pathQualifier: PathQualifier, session: SparkSession)
           case Some(sourceType) if sourceType == "mongo" || sourceType.isInstanceOf[com.mongodb.spark.sql.DefaultSource] =>
             asMongoDBWriteCommand(cmd)
 
+          case Some(sourceType) if sourceType == "elasticsearch" || sourceType.isInstanceOf[org.elasticsearch.spark.sql.DefaultSource15] =>
```

Comment:
I just noticed the usage if `.isInstanceOf` here as well as for Mongo and Cassandra cases. Although it works in tests, but will break in production due to missing classes in runtime. `.isInstanceOf[A]` requires class `A` to be loaded into JVM.
Instead we need to utilize `SafeTypeMatchingExtractor` - the same approach that is used for Excel type checking (see line #60). Similar to what you do in the `ReadCommandExtractor` class.

Reply:
Ok I'll take a look and try to correct all 3